base_layer/wallet/src/output_manager_service/storage/database.rs

```diff
@@ -0,0 +1,376 @@
+    }
+
+    pub fn add_unspent_output(&mut self, output: UnblindedOutput) -> Result<(), OutputManagerStorageError> {
+        self.db.write(WriteOperation::Insert(DbKeyValuePair::UnspentOutput(
+            output.spending_key.clone(),
+            Box::new(output),
+        )))?;
+
+        Ok(())
+    }
+
+    pub fn get_balance(&self) -> Result<MicroTari, OutputManagerStorageError> {
+        if let DbValue::UnspentOutputs(uo) =
+            self.db
+                .fetch(&DbKey::UnspentOutputs)?
+                .ok_or(OutputManagerStorageError::UnexpectedResult(
+                    "Unspent Outputs cannot be retrieved".to_string(),
+                ))?
+        {
+            Ok(uo.iter().fold(MicroTari::from(0), |acc, x| acc + x.value))
+        } else {
+            Err(OutputManagerStorageError::UnexpectedResult(
+                "Unexpected result from database backend".to_string(),
+            ))
+        }
+    }
+
+    pub fn add_pending_transaction_outputs(
+        &mut self,
+        pending_transaction_ouputs: PendingTransactionOutputs,
+    ) -> Result<(), OutputManagerStorageError>
+    {
+        self.db
+            .write(WriteOperation::Insert(DbKeyValuePair::PendingTransactionOutputs(
+                pending_transaction_ouputs.tx_id.clone(),
+                Box::new(pending_transaction_ouputs),
+            )))?;
+
+        Ok(())
+    }
+
+    pub fn fetch_pending_transaction_outputs(
+        &self,
+        tx_id: TxId,
+    ) -> Result<PendingTransactionOutputs, OutputManagerStorageError>
+    {
+        fetch!(self, tx_id, PendingTransactionOutputs)
+    }
+
+    /// This method is called when a pending transaction is confirmed. It moves the `outputs_to_be_spent` and
+    /// `outputs_to_be_received` from a `PendingTransactionOutputs` record into the `unspent_outputs` and
+    /// `spent_outputs` collections.
+    pub fn confirm_pending_transaction_outputs(&mut self, tx_id: TxId) -> Result<(), OutputManagerStorageError> {
+        let pending_tx = fetch!(self, tx_id.clone(), PendingTransactionOutputs)?;
+
+        // Add Spent outputs
+        for o in pending_tx.outputs_to_be_spent.iter() {
+            self.db.write(WriteOperation::Insert(DbKeyValuePair::SpentOutput(
+                o.spending_key.clone(),
+                Box::new(o.clone()),
+            )))?;
+        }
+
+        // Add Unspent outputs
+        for o in pending_tx.outputs_to_be_received.iter() {
+            self.db.write(WriteOperation::Insert(DbKeyValuePair::UnspentOutput(
+                o.spending_key.clone(),
+                Box::new(o.clone()),
+            )))?;
+        }
+
+        let _ = self
+            .db
+            .write(WriteOperation::Remove(DbKey::PendingTransactionOutputs(tx_id)))?;
+
+        Ok(())
+    }
+
+    /// This method is called when a transaction is built to be sent. It will encumber unspent outputs against a pending
+    /// transaction
+    pub fn move_unspent_outputs_to_pending_transaction(
```

Comment:
Similarly here. `reserve_utxos` or `mark_utxos_pending` describes the intent, rather than the _how_, which is up to the backend.

Reply:
Understood